lib/doorkeeper/request.rb

```diff
@@ -4,19 +4,22 @@ module Doorkeeper
   module Request
     class << self
       def authorization_strategy(response_type)
+        raise Errors::MissingRequiredParameter, :response_type if response_type.blank?
+
         get_strategy(response_type, authorization_response_types)
-      rescue NameError
-        raise Errors::InvalidAuthorizationStrategy
+      rescue StandardError
+        build_strategy_class("error_response_type")
```

Comment:
Introduce `ErrorResponseType` request for `authorization request` in `authorization_code` & `implicit` flow.

Because when `Errors::InvalidAuthorizationStrategy` happens, response is `ErrorResponse`, will be handle at [here](https://github.com/doorkeeper-gem/doorkeeper/blob/99f83f0dd88a215eb5b5f815f4b2a9e07caf9b5d/lib/doorkeeper/helpers/controller.rb#L48). AuthorizationController will check if that response is redirectable? by https://github.com/doorkeeper-gem/doorkeeper/blob/99f83f0dd88a215eb5b5f815f4b2a9e07caf9b5d/lib/doorkeeper/oauth/error_response.rb#L42-L45

So in this case ErrorResponse is always redirectable even `redirect_uri` is invalid because the `name` of error is always be `unsupported_response_type`